triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -198,40 +197,33 @@ object Server {
       new AkkaExecutionSequencerPool("TriggerService")(untypedSystem)
     implicit val dateTimeFormat: RootJsonFormat[LocalDateTime] = LocalDateTimeJsonFormat
 
-    def startTrigger(
-        ctx: ActorContext[Server.Message],
-        token: (Jwt, JwtPayload),
-        trigger: Trigger,
-        triggerName: Identifier,
-        ledgerConfig: LedgerConfig,
-        maxInboundMessageSize: Int,
-        maxFailureNumberOfRetries: Int,
-        failureRetryTimeRange: Duration,
-    ): JsValue = {
-      val jwt: Jwt = token._1
-      val jwtPayload: JwtPayload = token._2
-      val party: Party = Party(jwtPayload.party);
-      val triggerInstance = UUID.randomUUID
-      val ref = ctx.spawn(
-        TriggerRunner(
-          new TriggerRunner.Config(
-            ctx.self,
-            triggerInstance,
-            triggerName,
-            jwt,
-            server.compiledPackages,
-            trigger,
-            ledgerConfig,
-            maxInboundMessageSize,
-            maxFailureNumberOfRetries,
-            failureRetryTimeRange,
-            party
+    def startTrigger(token: (Jwt, JwtPayload), triggerName: Identifier): Either[String, JsValue] = {
+      for {
+        trigger <- Trigger.fromIdentifier(server.compiledPackages, triggerName).right
```

Comment:
I don’t think you need the `.right` here. `Either` has been right-biased for a while.